OpenSSL/crypto.py

```diff
@@ -1481,9 +1507,10 @@ def __init__(self):
 
     def add_cert(self, cert):
         """
-        Adds the certificate :py:data:`cert` to this store.
+        Adds a trusted certificate to this store.
```

Comment:
This isn't entirely correct. It is also used to add intermediate certificates (at least in your tests), which are not 'trusted' unless linked to a trusted self signed root certificate.

If X509_V_FLAG_PARTIAL_CHAINS is specified, it would be true, but that has other dangers. Currently the API to add 'chain' certificates to the StoreCTX that aren't trusted isn't used.